code/game/machinery/autolathe.dm

```diff
@@ -59,171 +35,195 @@
 		return
 
 	if(shocked && !(machine_stat & NOPOWER))
-		shock(user,50)
+		shock(user, 50)
+
 	ui = SStgui.try_update_ui(user, src, ui)
+
 	if(!ui)
 		ui = new(user, src, "Autolathe")
 		ui.open()
 
+/obj/machinery/autolathe/ui_static_data(mob/user)
+	var/list/data = list()
+
+	data["designs"] = handle_designs(stored_research.researched_designs)
+
+	return data
+
 /obj/machinery/autolathe/ui_data(mob/user)
 	var/list/data = list()
+
 	data["materials"] = list()
+
 	var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
+
 	data["materialtotal"] = materials.total_amount
 	data["materialsmax"] = materials.max_amount
-	data["categories"] = categories
-	data["designs"] = list()
 	data["active"] = busy
+	data["materials"] = materials.ui_data()
 
-	for(var/mat_id in materials.materials)
-		var/datum/material/M = mat_id
-		var/mineral_count = materials.materials[mat_id]
-		var/list/material_data = list(
-			name = M.name,
-			mineral_amount = mineral_count,
-			matcolour = M.greyscale_colors,
-		)
-		data["materials"] += list(material_data)
-	if(selected_category != "None" && !length(matching_designs))
-		data["designs"] = handle_designs(stored_research.researched_designs, TRUE)
-	else
-		data["designs"] = handle_designs(matching_designs, FALSE)
 	return data
 
-/obj/machinery/autolathe/proc/handle_designs(list/designs, categorycheck)
+/obj/machinery/autolathe/proc/handle_designs(list/designs)
 	var/list/output = list()
-	for(var/v in designs)
-		var/datum/design/D = categorycheck ? SSresearch.techweb_design_by_id(v) : v
-		if(categorycheck)
-			if(!(selected_category in D.category))
-				continue
+
+	var/datum/asset/spritesheet/research_designs/spritesheet = get_asset_datum(/datum/asset/spritesheet/research_designs)
+	var/size32x32 = "[spritesheet.name]32x32"
+
+	for(var/design_id in designs)
+		var/datum/design/design = SSresearch.techweb_design_by_id(design_id)
+
 		var/unbuildable = FALSE // we can't build the design currently
 		var/m10 = FALSE // 10x mult
 		var/m25 = FALSE // 25x mult
 		var/m50 = FALSE // 50x mult
 		var/m5 = FALSE // 5x mult
 		var/sheets = FALSE // sheets or no?
-		if(disabled || !can_build(D))
+
+		if(disabled || !can_build(design))
 			unbuildable = TRUE
+
 		var/max_multiplier = unbuildable ? 0 : 1
-		if(ispath(D.build_path, /obj/item/stack))
+
+		if(ispath(design.build_path, /obj/item/stack))
 			sheets = TRUE
+
 			if(!unbuildable)
```

Comment:
if this is possibly set by the power being out, what could this imply?

Reply:
Literally decides if a design should be built after all checks have ran. Also 1:1 with TG lathe logic.